RxSwiftWorkshop/Flights/ViewModel/FlightsViewModel.swift

```diff
@@ -18,22 +24,24 @@ final class FlightsViewModel: FlightsDisplayable {
     let iataCallable: IATACallable
 
     let flights: Variable<[Flight]>
-    private lazy var flightsCall: Observable<[Flight]> = {
-        self.schipolCallable
-            .flights()
-            .flatMap { flights in self.updateAirports(for: flights) }
-            .do(onError: { print($0) })
-            .catchErrorJustReturn([])
-    }()
 
     init(schipolCallable: SchipolCallable, iataCallable: IATACallable) {
         self.schipolCallable = schipolCallable
         self.iataCallable = iataCallable
         flights = Variable([])
-        flightsCall.bind(to: flights).disposed(by: disposeBag)
     }
 
     func refresh() -> Disposable {
+        return refresh(with: nil)
+    }
+
+    func refresh(with filter: FlightsFilter?) -> Disposable {
+        let flightsCall = schipolCallable
+            .flights(with: filter)
+            .flatMap { flights in self.updateAirports(for: flights) }
+            .do(onError: { print($0) })
+            .catchErrorJustReturn([])
```

Comment:
Yeah, one thing that moving the subscription call and binding to the ViewModel is that it makes it difficult for a ViewController to handle the errors. Would require some additional channel for error handling in production app, but for workshop app it can stay like this.